### ISSUE TITLE
Need to support Poisson, Gamma, and Tweedie regressions from LightGbm

### ISSUE THREAD

**Issue by wschin** [dotnet/machinelearning#1997]
As mentioned in #1424, for LightGbm regression, ML.NET interface only exposes L2-loss function to users.

**Comment by shauheen** [comment]
Thanks @wschin , @singlis can you please take a look and see how much work this is. We may want to postpone this for after v1.

**Comment by singlis** [comment]
@shauheen, here is the work that I see:
1) Add a regression enumerated type that includes Regression, Poisson, Gamma and Tweedie to our LightGBMRegression class. If not specified, it will default to Regression (which is what is currently used)
2)  Add tests that confirm that:
    1) This argument is being set correctly when passing to LigthGBM.
    2)  The output we produce is the same as LightGBM (i.e. run the test with ml.net, run the same test with LightGBM only and compare the results.) We have a test that already does this and I can borrow code from here. 

To account for the comment in #1424:
> As I mentioned above, for the Poisson, Gamma, and Tweedie regressions, the raw output of the tree needs to be made positive with a call to `Math.Exp`.

We will need to add a predictor to apply an exponential operation. This I am still looking into to get further details.

**Comment by singlis** [comment]
@shauheen - sorry for the delay. 
We need to add an expoenential calibrator for the exponential operation. 
I don't think this work would take long to do - it is not directly API related so it could be postponed to post 1.0.

**Comment by petterton** [comment]
@singlis, any chance this would be added soon?
